client/src/components/Layout/Login/Login.js

```diff
@@ -13,10 +13,11 @@ class Login extends Component {
   componentDidMount() {
     const { history } = this.props;
     axios.get("/api/check").then(({ data }) => {
-      const { success} = data;
-
-      if (success) return history.push("/home");
-    
+      const { success,userId} = data;
+      if (success)  {
+        this.setState({userId: userId})
```

Comment:
do you really need to put this into the state? Seems like it is not used anywhere (unless I'm missing sth)